agent/taskresource/envFiles/envfile.go

```diff
@@ -0,0 +1,198 @@
+package envFiles
+
+import (
+	"sync"
+	"time"
+
+	apicontainer "github.com/aws/amazon-ecs-agent/agent/api/container"
+	"github.com/aws/amazon-ecs-agent/agent/credentials"
+	resourcestatus "github.com/aws/amazon-ecs-agent/agent/taskresource/status"
+	"github.com/aws/amazon-ecs-agent/agent/s3/factory"
+	"github.com/cihub/seelog"
+	"github.com/aws/amazon-ecs-agent/agent/s3"
+	"path/filepath"
+	"strings"
+	"github.com/pkg/errors"
+	"github.com/aws/amazon-ecs-agent/agent/utils/oswrapper"
+
+	"github.com/aws/amazon-ecs-agent/agent/utils/ioutilwrapper"
+	"github.com/aws/amazon-ecs-agent/agent/taskresource"
+	"github.com/aws/amazon-ecs-agent/agent/api/task/status"
+)
+
+const (
+	envFileDirPath = "envfiles"
```

Comment:
in the design doc we didn't mention dedicated dir for env files, did something change?

Reply:
it did - "S3 env files will be downloaded to `/var/lib/ecs/data/envfiles/cluster_name/task_id/${s3filename.env}`"